app/org/sagebionetworks/bridge/models/accounts/UserSessionInfo.java

```diff
@@ -1,93 +1,91 @@
+    private static final String SESSION_TOKEN = "sessionToken";
+    private static final String AUTHENTICATED = "authenticated";
+    private static final String USERNAME = "username";
+    private static final String USER_SESSION_INFO = "UserSessionInfo";
+    private static final String TYPE = "type";
+    
+    private static final String PRODUCTION = "production";
+    private static final String STAGING = "staging";
+    private static final String DEVELOP = "develop";
+    private static final String LOCAL = "local";
+    
+    private static final BridgeObjectMapper MAPPER = BridgeObjectMapper.get();
+    private static final Set<String> PROHIBITED_FIELD_NAMES = Sets.newHashSet(TYPE, ENCRYPTED_HEALTH_CODE, CONSENT_HISTORIES);
+    private static final Map<Environment,String> ENVIRONMENTS = new ImmutableMap.Builder<Environment,String>()
+            .put(Environment.LOCAL, LOCAL)
+            .put(Environment.DEV, DEVELOP)
+            .put(Environment.UAT, STAGING)
+            .put(Environment.PROD, PRODUCTION).build();
 
-    public boolean isAuthenticated() {
-        return authenticated;
-    }
-    public Map<SubpopulationGuid,ConsentStatus> getConsentStatuses() {
-        return consentStatuses;
-    }
-    public boolean isConsented() {
-        return ConsentStatus.isUserConsented(consentStatuses);
-    }
-    public boolean isSignedMostRecentConsent() {
-        return ConsentStatus.isConsentCurrent(consentStatuses);
-    }
-    public SharingScope getSharingScope() {
-        return sharingScope;
-    }
-    public boolean isDataSharing() {
-        return (sharingScope != SharingScope.NO_SHARING);
-    }
-    public String getSessionToken() {
-        return sessionToken;
-    }
     /**
-     * Provided for API compatibility, this is always the email address of the account. 
-     * @deprecated
+     * Collapse StudyParticipant properties into the UserSessionInfo JSON so that we 
+     * do not need to copy these info the session when we update StudyParticipant. 
+     * Unfortunately this makes for some gross JSON processing. This can be moved 
+     * to a deserializer. 
      */
-    public String getUsername() {
-        return email;
-    }
-    public String getEmail() {
-        return email;
-    }
-    public String getEnvironment() {
-        return environment;
-    }
-    public Set<Roles> getRoles() {
-        return roles;
-    }
-    public Set<String> getDataGroups() {
-        return dataGroups;
-    }
-    public String getId() {
-        return id;
+    public static JsonNode toJSON(UserSession session) {
+        ObjectNode node = JsonNodeFactory.instance.objectNode();
+        node.put(AUTHENTICATED, session.isAuthenticated());
+        node.put(SESSION_TOKEN, session.getSessionToken());
+        node.put(ENVIRONMENT, ENVIRONMENTS.get(session.getEnvironment()));
+        node.put(DATA_SHARING, session.getParticipant().getSharingScope() != SharingScope.NO_SHARING);
```

Comment:
Do we want just the isDataSharing boolean? Or do we also want the sharing scope as well?